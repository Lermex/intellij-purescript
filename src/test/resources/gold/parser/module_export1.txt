Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Hello')
        PsiElement(whitespace)(' ')
      PsiElement(()('(')
      PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
        PSValueRefImpl(ValueRef)
          PSIdentifierImpl(identifier)
            PsiElement(identifier)('main')
      PsiElement(,)(',')
      PsiElement(whitespace)(' ')
      PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
        PSImportModuleNameImpl(ImportModuleName)
          PsiElement(module)('module')
          PsiElement(whitespace)(' ')
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('A')
      PsiElement(,)(',')
      PsiElement(whitespace)(' ')
      PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
        PSProperNameImpl(ProperName)
          PsiElement(proper name)('Label')
      PsiElement(,)(',')
      PsiElement(whitespace)(' ')
      PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
        PSProperNameImpl(ProperName)
          PsiElement(proper name)('User')
        PsiElement(()('(')
        PsiElement(..)('..')
        PsiElement())(')')
      PsiElement(,)(',')
      PsiElement(whitespace)(' ')
      PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
        PSProperNameImpl(ProperName)
          PsiElement(proper name)('Core')
        PsiElement(()('(')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Main')
        PsiElement(,)(',')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Data')
        PsiElement())(')')
      PsiElement())(')')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')