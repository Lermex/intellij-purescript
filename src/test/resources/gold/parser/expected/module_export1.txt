Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Hello')
    PSModuleExportsImpl(MODULE_EXPORTS)
      PsiElement(PSTokenType.()('(')
      PSModuleExportImpl(MODULE_EXPORT)
        PSModuleExportIdImpl(MODULE_EXPORT_ID)
          PsiElement(PSTokenType.ID)('main')
        PsiElement(PSTokenType.,)(',')
        PSModuleExportImpl(MODULE_EXPORT)
          PSModuleExportModuleNameImpl(MODULE_EXPORT_MODULE_NAME)
            PsiElement(PSTokenType.module)('module')
            PsiElement(PSTokenType.PROPER_NAME)('A')
          PsiElement(PSTokenType.,)(',')
          PSModuleExportImpl(MODULE_EXPORT)
            PSModuleExportTypeImpl(MODULE_EXPORT_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('Label')
            PsiElement(PSTokenType.,)(',')
            PSModuleExportImpl(MODULE_EXPORT)
              PSModuleExportTypeConstructorImpl(MODULE_EXPORT_TYPE_CONSTRUCTOR)
                PSModuleExportTypeImpl(MODULE_EXPORT_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('User')
                PsiElement(PSTokenType.()('(')
                PsiElement(PSTokenType...)('..')
                PsiElement(PSTokenType.))(')')
              PsiElement(PSTokenType.,)(',')
              PSModuleExportImpl(MODULE_EXPORT)
                PSModuleExportTypeConstructorImpl(MODULE_EXPORT_TYPE_CONSTRUCTOR)
                  PSModuleExportTypeImpl(MODULE_EXPORT_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('Core')
                  PsiElement(PSTokenType.()('(')
                  PSModuleExportTypeConstructorItemImpl(MODULE_EXPORT_TYPE_CONSTRUCTOR_ITEM)
                    PsiElement(PSTokenType.PROPER_NAME)('Main')
                    PsiElement(PSTokenType.,)(',')
                    PsiElement(PSTokenType.PROPER_NAME)('Data')
                  PsiElement(PSTokenType.))(')')
      PsiElement(PSTokenType.))(')')
    PsiElement(PSTokenType.where)('where')